pkg/reconciler/instances/istio/reset/data/gatherer_test.go

```diff
@@ -72,8 +74,10 @@ func Test_Gatherer_GetPodsWithDifferentImage(t *testing.T) {
 		pods.Items = []v1.Pod{
 			*podWithExpectedImage,
 			*podWithExpectedImageTerminating,
+			*podWithExpectedImagePending,
```

Comment:
I had a look at `should get one pod from the list` test case - could you please add one more assertion to verify which exactly pod was returned in the list? There are many of them so we should be sure that appropriate one was returned.